src/lib/config/NetworkConfigProvider.tsx

```diff
@@ -1,29 +1,45 @@
 import { NetworkConfigContext } from '.';
+import { SupportedNetworkContext } from './useSupportedNetwork';
 import { useNetwork } from 'wagmi';
-import { networkConfigs } from './networkConfig';
-import { NetworkConfig } from './networkConfigType';
+import { networkConfigs } from './networkConfigs';
 
 export function NetworkConfigProvider({ children }: { children: React.ReactNode }) {
   const { chain } = useNetwork();
-  const networkConfig: NetworkConfig = !!chain
-    ? networkConfigs[chain.id]
-    : // fallback to empty config if there is no connected wallet
-      {
-        chainId: 0,
-        networkName: '',
-        networkShortName: '',
-        sarcoTokenAddress: '',
-        diamondDeployAddress: '',
-        explorerUrl: '',
-        explorerApiKey: '',
-        bundlr: {
-          currencyName: '',
-          nodeUrl: '',
-          providerUrl: '',
-        },
-      };
+
+  const networkConfig =
+    !!chain && networkConfigs[chain.id]
+      ? networkConfigs[chain.id]
+      : {
+          chainId: 0,
+          networkName: '',
+          networkShortName: '',
+          sarcoTokenAddress: '',
+          diamondDeployAddress: '',
+          explorerUrl: '',
+          explorerApiKey: '',
+          bundlr: {
+            currencyName: '',
+            nodeUrl: '',
+            providerUrl: '',
+          },
+        };
+
+  const supportedChainIds =
+    process.env.REACT_APP_SUPPORTED_CHAIN_IDS?.split(',').map(id => parseInt(id)) || [];
+
+  const isSupportedChain = supportedChainIds.includes(networkConfig.chainId);
+
+  const supportedNetworkNames = Object.values(networkConfigs)
+    .filter(config => supportedChainIds.includes(config.chainId))
+    .map(config => config.networkShortName);
 
   return (
-    <NetworkConfigContext.Provider value={networkConfig}>{children}</NetworkConfigContext.Provider>
+    <NetworkConfigContext.Provider value={networkConfig}>
+      <SupportedNetworkContext.Provider
```

Comment:
Add an additional context to return `SuppportedNetwork` properties. Attempted to add directly to the NetworkConfig, but in the end separating them made more sense.